baybe/surrogates/base.py

```diff
@@ -124,60 +134,80 @@ def _make_target_scaler() -> OutcomeTransform | None:
 
     def _make_input_scaler(self, searchspace: SearchSpace) -> ColumnTransformer:
         """Make the input scaler for transforming computational dataframes."""
-        from sklearn.compose import make_column_transformer
-
-        # Create the composite scaler from the parameter-wise scaler objects
-        transformers = [
-            (
-                "passthrough" if (s := self._make_parameter_scaler(p)) is None else s,
-                [c for c in p.comp_rep_columns if c in searchspace.comp_rep_columns],
-            )
-            for p in searchspace.parameters
-        ]
-        scaler = make_column_transformer(*transformers, verbose_feature_names_out=False)
-
-        scaler.fit(searchspace.comp_rep_bounds)
+        # Create a composite scaler from parameter-wise scaler objects
+        mapping: dict[tuple[int, ...], InputTransform] = {}
+        for p in searchspace.parameters:
+            idxs = searchspace.get_comp_rep_parameter_indices(p.name)
+            factory = self._make_parameter_scaler_factory(p)
+            if factory is None:
+                continue
+            transformer = factory(len(idxs))
+            mapping[idxs] = transformer
+        scaler = ColumnTransformer(mapping)
+
+        # Fit the scaler to the parameter bounds
+        scaler.fit(to_tensor(searchspace.comp_rep_bounds))
 
         return scaler
```

Comment:
The difference is that the output scaler is an actual `botorch` transform (i.e. using `eval`/`train`) whereas the input scaler is our new custom `ColumnTransformer`, which I wrote closer to the `sklearn` style (`fit`/`transform`). So internally, `fit` sets the individual transformers to `train` while `transform` sets them to `eval`. 

This is just an arbitrary design. If you you prefer that we switch to the other variant, we can also do so...

Reply:
the difference is clear, I just wanted to point out that at the end of their respective functions, the input scaler is in `train` mode and the output one in `eval` mode

Reply:
Ah OK, now I see what you mean. But not causing problems, right? Or would you prefer that we automatically put the input scaler in `eval` modes after fitting?

Reply:
The fit/transform interface is safer because you don't have to remember putting the mode etc
to make the other a bit safer i guess its indeed better to do the fitting like 
```
x.train()
x(training_data)
x.eval()
```
I just want it to be safe and if you say so thats fine

Reply:
Have a look how you like the new version, which now makes the check explicit. If OK, feel free to resolve